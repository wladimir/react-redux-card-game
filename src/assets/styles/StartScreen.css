.intro-text {
  font-family: "Lobster", sans-serif;
  margin-bottom: 10px;

  font-size: 72px;
  background: -webkit-linear-gradient(rgb(230, 255, 1), rgb(173, 156, 1));
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;
}

.button {
  border: none;
  color: black;
  background-color: #c8faca;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  width: 100%;
  border-radius: 15px / 45px;

  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;

  margin-bottom: 15px;
}

.button:active {
  background-color: #3e8e41;
  box-shadow: 0 5px #666;
  transform: translateY(4px);
}

.button:hover {
  background-color: #4caf50;
  color: white;
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (min-device-width: 320px) and (max-device-width: 667px) {
  .intro-text {
    font-size: 25px;
  }

  .button {
    margin-bottom: 5px;
  }
}
