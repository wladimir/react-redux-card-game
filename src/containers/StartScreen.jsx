import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Button from "../components/Button";
import { startGame } from "../actions";
import { Link } from "react-router-dom";
import "../assets/styles/StartScreen.css";

const StartScreen = ({ startGame }) => {
  return (
    <div>
      <div className="center">
        <div className="intro-text">Select number of players</div>
        <Link to="/game">
          <Button text={"2 players"} onClick={() => startGame(2)} />
        </Link>
        <Link to="/game">
          <Button text={"3 players"} onClick={() => startGame(3)} />
        </Link>
        <Link to="/game">
          <Button text={"4 players"} onClick={() => startGame(4)} />
        </Link>
      </div>
    </div>
  );
};

const mapDispatchToProps = dispatch => {
  return {
    startGame: playerCount => dispatch(startGame(playerCount))
  };
};

StartScreen.propTypes = {
  startGame: PropTypes.func.isRequired
};

export default connect(
  null,
  mapDispatchToProps
)(StartScreen);
