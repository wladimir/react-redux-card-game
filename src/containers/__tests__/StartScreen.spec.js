import React from "react";
import { StartScreen } from "../StartScreen";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";

describe("StartScreen", () => {
  it("should contain title and 3 buttons", () => {
    const component = renderer.create(
      <MemoryRouter>
        <StartScreen startGame={() => {}} />
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
