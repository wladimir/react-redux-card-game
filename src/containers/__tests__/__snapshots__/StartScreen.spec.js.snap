// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreen should contain title and 3 buttons 1`] = `
<div
  className="center"
>
  <p
    className="intro-text"
  >
    Select number of players
  </p>
  <a
    href="/game"
    onClick={[Function]}
  >
    <button
      className="button"
      onClick={[Function]}
    >
      2 players
    </button>
  </a>
  <a
    href="/game"
    onClick={[Function]}
  >
    <button
      className="button"
      onClick={[Function]}
    >
      3 players
    </button>
  </a>
  <a
    href="/game"
    onClick={[Function]}
  >
    <button
      className="button"
      onClick={[Function]}
    >
      4 players
    </button>
  </a>
</div>
`;
